presentations/vocabulary_types_2023_11_8.md

```diff
@@ -0,0 +1,241 @@
+---
+marp: true
+theme: default
+paginate: true
+size: 16:9
+footer: https://github.com/jbcoe/value_types
+---
+
+# Vocabulary Types for Composite Class Design
+
+_Jonathan B. Coe, Antony Peacock & Sean Parent_
+
+## ISO/IEC JTC1 SC22 WG21 Programming Language C++
+
+## [P3019r0](https://wg21.link/p3019r0)
+
+### Kona Hawaii 2023-11-8
+
+
+---
+
+# Introduction
+
+We propose adding two new class templates to the C++ Standard Library.
+
+```c++
+template <typename T, typename Allocator = std::allocator<T>>
+class indirect;
+```
+
+
+```c++
+template <typename T, typename Allocator = std::allocator<T>>
+class polymorphic;
+```
+
+These fill an existing gap in the suite of existing standard library vocabulary types.
+
+---
+
+# Vocabulary Types
+
+We refer to standard library types like `std::array`, `std::map`, `std::optional`,  `std::variant` and `std::vector` as _vocabulary types_.
+
+We'd anticipate that an arbitrary piece of C++ library or application code would make use of some of these types.
+
+Standardizing vocabulary types is important as it allows different libraries to easily interoperate and for users to build applications.
+
+The standard library contains other, non-vocabulary types to do specific jobs like interacting with the filesystem, formatting text for output, or dealing with concurrency.
+
+We probably want to standardize both sorts of library type.
+
+---
+
+# Composite classes
+
+We can define composite classes: classes with other class instances as member data.
```

Comment:
We define composite classes as: classes with other class instances as member data.

Reply:
I went with "We define composite classes as classes with other class instances as member data."